src/components/Project/ProjectBasics/ProjectForm/ProjectForm.tsx

```diff
@@ -54,6 +56,116 @@ const ProjectForm = () => {
     when: isDirty,
   });
 
+  const getFinanceYearName = (finances: IProjectFinances, year: number) => {
+    const index = year - finances.year;
+    if (index < 3) {
+      return "budgetProposalCurrentYearPlus" + index as keyof IProjectFinances;
+    }
+    else {
+      return "preliminaryCurrentYearPlus" + index as keyof IProjectFinances;
+    }
+  }
+
+  const moveBudgetForwards = (finances: IProjectFinances, previousStartYear: number, startYear: number) => {
+    let financesCopy = finances
+    const numberOfYears = startYear - previousStartYear;
+    let budgetToMove = 0.00;
+    for (let i=0; i<numberOfYears; ++i) {
+      const financeYearName = getFinanceYearName(finances, previousStartYear + i);
+      const financeValue = finances[financeYearName];
+      budgetToMove += parseFloat(financeValue as string);
+      financesCopy = { ...financesCopy, [financeYearName]: "0.00"};
+    }
+    const startYearName = getFinanceYearName(finances, startYear);
+    const newBudget = (parseFloat(finances[startYearName] as string) || 0.00) + budgetToMove;
+
+    financesCopy = { ...financesCopy, [startYearName]: newBudget.toFixed(2)}
+    return financesCopy;
+  }
+
+  const moveBudgetBackwards = (finances: IProjectFinances, previousEndYear: number, endYear: number) => {
```

Comment:
Maybe you could add some little comments of what happens here in this function so that it would be quicker to read it through?  :smile:

Reply:
Added comments 👍